File: src/main/java/com/dingdongdeng/coinautotrading/common/logging/LoggingUtils.java
package com.dingdongdeng.coinautotrading.common.logging;

import java.util.Map;
import java.util.Objects;
import java.util.UUID;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;
import org.slf4j.MDC;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class LoggingUtils {

    public static void track() {
        clear();
        put("trackingId", UUID.randomUUID().toString().replaceAll("-", "").substring(0, 10));
    }

    public static void put(String key, String value) {
        MDC.put(key, value);
    }

    public static String get(String key) {
        return MDC.get(key);
    }

    public static Map<String, String> getLogData() {
        return MDC.getCopyOfContextMap();
    }

    public static void setLogData(Map<String, String> contextMap) {
        if (Objects.nonNull(contextMap)) {
            MDC.setContextMap(contextMap);
        }
    }

    public static void clear() {
        MDC.clear();
    }

}

File: app-service/src/main/kotlin/com/dingdongdeng/autotrading/infra/common/log/LoggingUtils.kt
package com.dingdongdeng.autotrading.infra.common.log

import org.slf4j.MDC
import java.util.Objects
import java.util.UUID

object LoggingUtils {
    private const val TRACE_ID = "traceId"
    fun trace() {
        clear()
        put(TRACE_ID, UUID.randomUUID().toString().replace("-".toRegex(), "").substring(0, 10))
    }

    fun put(key: String, value: String) {
        MDC.put(key, value)
    }

    fun get(key: String): String {
        return MDC.get(key)
    }

    fun getTraceId(): String {
        return MDC.get(TRACE_ID)
    }

    fun getLogData(): Map<String, String> {
        return MDC.getCopyOfContextMap()
    }

    fun setLogData(contextMap: Map<String, String>) {
        if (Objects.nonNull(contextMap)) {
            MDC.setContextMap(contextMap)
        }
    }

    fun clear() {
        MDC.clear()
    }
}
